### PR TITLE
Ft test

### PR DESCRIPTION
**Pull request by enoskyeza**
Features that were stated in the token are ready for review.

### CODE REVIEW

File: lab_management/admin.py [hunk 1]
```diff
@@ -0,0 +1,6 @@
+from cgi import test
+from django.contrib import admin
+from . import models
```
Comment (Kyeza) [inline lab_management/admin.py:3]: change this import to
`from lab_management.models import Test `

File: lab_management/admin.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+from cgi import test
```
Comment (Kyeza) [inline lab_management/admin.py:1]: remove this redundant code

File: lap_pro/urls.py [hunk 3]
```diff
@@ -18,5 +18,6 @@
 
 urlpatterns = [
     path('patient/', include('patient_management.urls')),
```
Comment (Kyeza) [inline lap_pro/urls.py:20]: rename these URLs to plural

- patients
- tests

**Review by Kyeza — CHANGES_REQUESTED** [review]

File: lab_management/views.py [hunk 4]
```diff
@@ -0,0 +1,40 @@
+from django.shortcuts import render
+from django.urls import reverse
+from .models import Test
```
Comment (Kyeza) [inline lab_management/views.py:3]: Use absolute imports

File: lab_management/views.py [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+from .forms import TestForm
```
Comment (Kyeza) [inline lab_management/views.py:4]: Use absolute imports

**Review by Kyeza — CHANGES_REQUESTED** [review]

File: lab_management/admin.py [hunk 6]
```diff
@@ -0,0 +1,5 @@
+from django.contrib import admin
+from .models import Test
```
[the remark this answers is not in the document]
Comment (enoskyeza) [reply lab_management/admin.py:2]: I had not understood what you meant, now that you say relative referencing I understand you well.

[the remark this answers is not in the document]
Comment (enoskyeza) [reply lab_management/admin.py:2]: but I have seen them using relative import in the django documentation more so when importing items from the same app.

[the remark this answers is not in the document]
Comment (Kyeza) [reply lab_management/admin.py:2]: Do what I have asked bro, it's better to use absolute

File: lab_management/models.py [hunk 7]
```diff
@@ -0,0 +1,37 @@
...
+    type = models.ForeignKey(SampleType, null=True, on_delete=models.SET_NULL)
+    collection_site = models.CharField(max_length=50)
+    date_of_collection = models.DateTimeField()
+    lab_reference = models.CharField(max_length=150)
+    date_of_result = models.DateTimeField()
+
+class TestRequest(BaseModel):
+    patient = models.ForeignKey(Patient, null=True, on_delete=models.SET_NULL)
+    test = models.ForeignKey(Test, null=True, on_delete=models.SET_NULL)
+    sample = models.ForeignKey(Sample, null=True, on_delete=models.SET_NULL)
```
Comment (Kyeza) [inline lab_management/models.py:36]: add `blank=True` as it will allow this field to be left blank in order to be added later

**Review by Kyeza — CHANGES_REQUESTED** [review]

File: lab_management/models.py [hunk 8]
```diff
@@ -0,0 +1,37 @@
...
+    name = models.CharField(max_length=50)
+
+    def __str__(self):
+        return self.name
+
+class Sample(BaseModel):
+    type = models.ForeignKey(SampleType, null=True, on_delete=models.SET_NULL)
+    collection_site = models.CharField(max_length=50)
+    date_of_collection = models.DateTimeField()
+    lab_reference = models.CharField(max_length=150)
```
[the remark this answers is not in the document]
Comment (enoskyeza) [reply lab_management/models.py:30]: Alright

**Review by Kyeza — APPROVED** [review]

**Pull request merged**